decision-records/008-reset-the-javascript-api.md

```diff
@@ -0,0 +1,72 @@
+# Reset the JavaScript API for our components
+
+## Decision
+
+We will mark all class and instance properties other than the constructor and the init method as deprecated as part of a v4.x release.
+
+In v5.0 we will then add JSDoc annotations to mark the deprecated properties as private.
```

Comment:
Added a comment below about "internal" things that we'd mark with [JSDoc `@protected`](https://jsdoc.app/tags-protected.html)

```mjs
class Accordion {
  #secret

  /** @protected */
  $module

  constructor ($module, secret) {
    this.$module = $module
    this.#secret = secret
  }
}

class SuperAccordion extends Accordion {
  constructor ($module) {
    super($module)
    
    // ✅ Protected things
    console.log(this.$module)
    
    // ❌ Private things
    console.log(this.#secret)
  }
}
```

Reply:
Similar to above, I don't think there's anything in this decision that'd prevent us from being able to adopt `@protected` in the future, where it makes sense to do so.